src/main/java/uk/gov/hmcts/reform/sendletter/logging/AppInsights.java

```diff
@@ -66,6 +70,40 @@ public Object trackDependency(ProceedingJoinPoint joinPoint, Dependency dependen
         }
     }
 
+    @Pointcut("@annotation(dependency)")
+    public void internalDependencyPointCut(InternalDependency dependency) {
+        // point cut definition
+    }
+
+    @Around("internalDependencyPointCut(dependency)")
+    public Object trackInternalDependency(
+        ProceedingJoinPoint joinPoint,
+        InternalDependency dependency
+    ) throws Throwable {
+        Instant start = Instant.now();
+
+        try {
+            Object proceed = joinPoint.proceed();
+
+            telemetry.trackEvent(
+                dependency.value(),
+                ImmutableMap.of("success", "true"),
+                ImmutableMap.of("timeTaken", (double) MILLIS.between(start, Instant.now()))
+            );
+
+            return proceed;
+        } catch (Throwable exception) {
+            telemetry.trackEvent(
+                dependency.value(),
+                ImmutableMap.of("success", "false"),
+                ImmutableMap.of("timeTaken", (double) MILLIS.between(start, Instant.now()))
+            );
+            telemetry.trackException((Exception) exception);
```

Comment:
not a part of this PR, but is this casting safe?

Reply:
If `Error` is being caught here then essentially it is a bad thing and should be thrown. No application should experience `Error`

Reply:
Actually, I think we should catch Exception instead of Throwable.

Reply:
We should. `Throwable` comes from `jointPoint.proceed()`. Oh wait, just realised I am throwing it anyway and marking method as throwable. I'll update those advices